activerecord/lib/active_record/relation.rb

```diff
@@ -374,7 +374,7 @@ def update_all(updates)
       stmt.key = table[primary_key]
 
       if joins_values.any?
-        @klass.connection.join_to_update(stmt, arel)
+        @klass.connection.join_to_update(stmt, arel, table[primary_key])
```

Comment:
Which is better `table[primary_key]` or `stmt.key` as 3rd argument ?
Before in `join_to_update` `key` was `stmt.key`, so I feel `stmt.key` is better .